modules/gapi/samples/pipeline_modeling_tool/utils.hpp

```diff
@@ -91,6 +93,26 @@ typename duration_t::rep timestamp() {
     return duration_cast<duration_t>(now.time_since_epoch()).count();
 }
 
+#define RETHROW_WITH_MSG_IF_FAILED(expr, msg)      \
+    try {                                          \
+        expr;                                      \
+    } catch (const std::exception& e) {            \
+        std::stringstream ss;                      \
+        ss << msg << "\n caused by: " << e.what(); \
+        throw std::logic_error(ss.str());          \
+    }                                              \
+
+template <typename K, typename V>
+void intersectMapWith(std::map<K, V>& target, const std::map<K, V>& second) {
+    for (auto&& item : second) {
+        auto it = target.find(item.first);
+        if (it != target.end()) {
+            throw std::logic_error("Met already existing key: " + item.first);
+        }
+        target.insert(item);
```

Comment:
In set theory, intersection is something different from this, isn't it?

Don't you just merge two maps into one with handling the "overwrites"?

Reply:
Yes, good point!

Reply:
Done